include/ghex/glue/atlas/atlas_user_concepts.hpp

```diff
@@ -0,0 +1,467 @@
+                const T& get(const index_t idx, const std::size_t level) const {
+                    return m_values(static_cast<std::size_t>(idx), level); // WARN: why std::size_t cast is needed here?
+                }
+
+                /** @brief multiple access set function, needed by GHEX in order to perform the unpacking.
+                 * WARN: it could be more efficient if the iteration space includes also the indexes on this domain;
+                 * in order to do so, iteration space needs to include an additional set of indexes;
+                 * for now, the needed indices are retrieved by looping over the whole doamin,
+                 * and filtering out all the indices by those on the desired remote partition.
+                 * @tparam IterationSpace iteration space type
+                 * @param is iteration space which to loop through in order to retrieve the coordinates at which to set back the buffer values
+                 * @param buffer buffer with the data to be set back*/
+                template <typename IterationSpace>
+                void set(const IterationSpace& is, const Byte* buffer) {
+                    for (index_t idx : is.local_index()) {
+                        for (std::size_t level = 0; level < is.levels(); ++level) {
+                            set(*(reinterpret_cast<const T*>(buffer)), idx, level);
+                            buffer += sizeof(T);
+                        }
+                    }
+                }
+
+                /** @brief multiple access get function, needed by GHEX in order to perform the packing
+                 * @tparam IterationSpace iteration space type
+                 * @param is iteration space which to loop through in order to retrieve the coordinates at which to get the data
+                 * @param buffer buffer to be filled*/
+                template <typename IterationSpace>
+                void get(const IterationSpace& is, Byte* buffer) const {
+                    for (index_t idx : is.local_index()) {
+                        for (std::size_t level = 0; level < is.levels(); ++level) {
+                            std::memcpy(buffer, &get(idx, level), sizeof(T));
+                            buffer += sizeof(T);
+                        }
+                    }
+                }
+
+                template<typename IndexContainer>
+                void pack(T* buffer, const IndexContainer& c, void*) {
+                    for (const auto& is : c) {
+                        get(is, reinterpret_cast<Byte*>(buffer));
+                    }
+                }
+
+                template<typename IndexContainer>
+                void unpack(const T* buffer, const IndexContainer& c, void*) {
+                    for (const auto& is : c) {
+                        set(is, reinterpret_cast<const Byte*>(buffer));
+                    }
+                }
+
+        };
+
+#ifdef __CUDACC__
+
+#define THREADS_PER_BLOCK 32
```

Comment:
the name of this macro is not very unique - it could potentially clash with other (user-defined) macros. Use a prefix GHEX and a more descriptive atlas-specific name.

Reply:
Ok, right. Done.